### ISSUE TITLE
False positive: ChainWrapping

### ISSUE THREAD

**Issue by mrclrchtr** [detekt/detekt#7577]
## Expected Behavior
```
  fun isDebugMode(): Boolean {
    return ManagementFactory.getRuntimeMXBean().inputArguments.any {
      it.contains("-agentlib:jdwp")
        || it.contains("-Xdebug")
        || it.contains("-Xrunjdwp")
    }
  }
```

Should not get detekted.

## Observed Behavior
Warning: `detekt - ChainWrapping: Line must not begin with "||"`

+ Auto Correction to:
```
  fun isDebugMode(): Boolean {
    return ManagementFactory.getRuntimeMXBean().inputArguments.any {
      it.contains("-agentlib:jdwp") ||
        it.contains("-Xdebug") ||
        it.contains("-Xrunjdwp")
    }
  }
```

## Your Environment
* Version of detekt used: 1.23.6
* Version of Gradle used (if applicable): 8.10
* Operating System and version: MacOS

**Comment by 3flex** [comment]
We wrap the ktlint rule. Can you please check if this is an issue using ktlint direct?

**Comment by cortinico** [comment]
As @3flex mentioned, this should be reported to https://github.com/pinterest/ktlint/

> Should not get detekted.

Also why not? I would argue that the formatted snippet is preferable because the first operator is inlined. Having the operator on the new line could lead to subtle bugs.

**Comment by mrclrchtr** [comment]
Ok, thanks for the answer. 

I have learned that it makes sense to write the `||` or other "connectors" in the next line. How should the line before know if another line follows? Even when making changes, you only change the corresponding line and not the one above it.

But yes, then this belongs in the ktlint project.
